### PR TITLE
Transform configResponse data

### PR DESCRIPTION
**Pull request by myang1010**
What (if any) features are you implementing?
- Updated `transformConfigResponse` function to handle potentially nested `configResponse` data.

Any other comments, questions, or concerns?
- Please let me know if it makes sense.

### CODE REVIEW

File: src/Components/Config/configHook.tsx [hunk 1]
```diff
@@ -1,14 +1,34 @@
 import { useEffect, useState } from 'react';
 import { configEndpoint, header } from '../../apiCalls';
-
 import { ConfigApiResponse } from '../../Types/Config';
 import { Config } from '../../Types/Config';
+import { FormattedMessage } from 'react-intl';
+
+// Recursive function
```
Comment (CalebPena) [inline src/Components/Config/configHook.tsx:7]:
```suggestion
// Recursively transform any object that has _label, and _default_message as keys into a FormattedMessage
```

File: src/Components/Config/configHook.tsx [hunk 2]
```diff
@@ -1,14 +1,34 @@
...
+  const config: Config = {};
+  for (const key in item) {
+    if (item.hasOwnProperty(key)) {
+      config[key] = transformItem(item[key]);
+    }
+  }
+  return config;
+}
 
+// Main function with recursion
```
Comment (CalebPena) [inline src/Components/Config/configHook.tsx:26]: I don't think that this comment is very helpful. What do you think about removing it.

File: src/Components/Config/configHook.tsx [hunk 3]
```diff
@@ -1,14 +1,34 @@
 import { useEffect, useState } from 'react';
 import { configEndpoint, header } from '../../apiCalls';
-
 import { ConfigApiResponse } from '../../Types/Config';
 import { Config } from '../../Types/Config';
+import { FormattedMessage } from 'react-intl';
+
+// Recursive function
+function transformItem(item: any): any {
```
[the remark this answers is not in the document]
Comment (myang1010) [reply src/Components/Config/configHook.tsx:8]: I'm still relative new to TypeScript. I'll get `Cannot find name 'unknow'.ts(2304)` error if I use `unknown` in my VSC.

[the remark this answers is not in the document]
Comment (CalebPena) [reply src/Components/Config/configHook.tsx:8]: I made a typo, should be `unknown`

[the remark this answers is not in the document]
Comment (myang1010) [reply src/Components/Config/configHook.tsx:8]: Lol, I should've noticed that!

**Pull request merged**